virtual_rainforest/models/soil/carbon.py

```diff
@@ -0,0 +1,203 @@
+"""The `soil.carbon` module.
+
+This module simulates the radiation soil carbon cycle for the Virtual Rainforest. At the
+moment only two pools are modelled, these are low molecular weight carbon (LMWC) and
+mineral associated organic matter (MAOM). More pools and their interactions will be
+added at a later date.
+"""
+
+import numpy as np
+from numpy.typing import NDArray
+
+from virtual_rainforest.core.logger import log_and_raise
+from virtual_rainforest.core.model import InitialisationError
+
+# from core.constants import CONSTANTS as C
+# but for meanwhile define all the constants needed here
+BINDING_WITH_PH = {
+    "slope": -0.186,  # unitless
+    "intercept": -0.216,  # unitless
+}
+"""From linear regression (Mayes et al. (2012))."""
+MAX_SORPTION_WITH_CLAY = {
+    "slope": 0.483,  # unitless
+    "intercept": 2.328,  # unitless
+}
+"""From linear regression (Mayes et al. (2012))."""
+MOISTURE_SCALAR = {
+    "coefficient": 30.0,  # unitless
+    "exponent": 9.0,  # unitless
+}
+"""Used in Abramoff et al. (2018), but can't trace it back to anything more concrete."""
+TEMP_SCALAR = {
+    "t_1": 15.4,  # C
+    "t_2": 11.75,  # unclear
+    "t_3": 29.7,  # unclear
+    "t_4": 0.031,  # unclear
+    "ref_temp": 30.0,  # C
+}
+"""Used in Abramoff et al. (2018), but can't trace it back to anything more concrete."""
+
+
+class SoilCarbonPools:
+    """Class containing the full set of soil carbon pools.
+
+    At the moment, only two pools are included. Functions exist for the transfer of
+    carbon between these pools, but not with either the yet to be implemented soil
+    carbon pools, other pools in the soil module, or other modules.
+
+    Attributes:
+        maom: Mineral associated organic matter pool
+        lmwc: Low molecular weight carbon pool
+    """
+
+    def __init__(self, maom: NDArray[np.float32], lmwc: NDArray[np.float32]) -> None:
+        """Initialise set of carbon pools."""
+
+        # Check that arrays are of equal size and shape
+        if maom.shape != lmwc.shape:
+            log_and_raise(
+                "Dimension mismatch for initial carbon pools!",
+                InitialisationError,
+            )
+
+        # Check that negative initial values are not given
+        if any(i < 0 for i in maom) or any(i < 0 for i in lmwc):
```

Comment:
I think this is better and likely faster as `if np.any(maom < 0) or np.any(lmwc < 0)`